### PR TITLE
[immer] update to 0.8.1

### PR DESCRIPTION
**Pull request by altschuler**
Immer has not made a new release (the project doesn't really use it), so the port version is just bumped.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by BillyONeal** [comment]
> Immer has not made a new release (the project doesn't really use it)

They issued a release in November; do you have a clear statement from upstream that they don't do releases anymore?

/cc @arximboldi

**Review by BillyONeal — CHANGES_REQUESTED** [review]
Looking for a clear statement from upstream on the versioning

**Comment by arximboldi** [comment]
We do releases, just rarely. If some feature is required and not released yet please open a ticket in Immer.

File: ports/immer/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "immer",
   "version": "0.8.0",
-  "port-version": 1,
+  "port-version": 2,
```
Comment (dg0yt) [inline ports/immer/vcpkg.json:4]: `0.8.0` means v0.8.0, not 5875f77.
If upstream releases aren't frequent enough, this port would need to switch to the `version-date` scheme.

Comment (altschuler) [reply ports/immer/vcpkg.json:4]: Yes, I'll change it when there's a new version tag for immer. I don't think we want to switch back to `version-date` for this, it was so before and was changed not too long ago.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
@altschuler Thanks for the update!

@arximboldi Thanks for the release!

**Pull request merged**